cli/src/generate/templates/cmakelists.txt

```diff
@@ -0,0 +1,53 @@
+cmake_minimum_required(VERSION 3.13)
+
+project(tree-sitter-PARSER_NAME
+        VERSION "0.0.1"
+        DESCRIPTION "CAMEL_PARSER_NAME grammar for tree-sitter"
+        HOMEPAGE_URL "https://github.com/tree-sitter/tree-sitter-PARSER_NAME"
+        LANGUAGES C)
+
+option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
+
+set(TREE_SITTER_ABI_VERSION ABI_VERSION_MAX CACHE STRING "Tree-sitter ABI version")
+if(NOT ${TREE_SITTER_ABI_VERSION} MATCHES "^[0-9]+$")
+    unset(TREE_SITTER_ABI_VERSION CACHE)
+    message(FATAL_ERROR "TREE_SITTER_ABI_VERSION must be an integer")
+endif(NOT ${TREE_SITTER_ABI_VERSION} MATCHES "^[0-9]+$")
+
+find_program(TREE_SITTER_CLI tree-sitter DOC "Tree-sitter CLI")
+
+add_custom_target(generate DEPENDS src/parser.c)
```

Comment:
I don't fully understand this. I can't see where this target is used. Is this meant to be used manually by a user?

Reply:
The main target depends on `src/parser.c` and will call this target if it is missing.

Reply:
I still suspect the target `generate` isn't needed and that only `add_custom_command` is required but I don't know how to test this.

Reply:
Maybe, but this allows users to run it manually.